src/promptflow/promptflow/_core/_errors.py

```diff
@@ -115,3 +115,38 @@ class MetaFileNotFound(GenerateMetaUserError):
 
 class MetaFileReadError(GenerateMetaUserError):
     pass
+
+
+class RunRecordNotFound(ValidationException):
+    pass
+
+
+class FlowOutputUnserializable(UserErrorException):
+    def __init__(self, *, output_name: str, module: str = None):
+        self._output_name = output_name
+        super().__init__(target=ErrorTarget.FLOW_EXECUTOR, module=module)
+
+    @property
+    def message_format(self):
+        if self.inner_exception:
+            return (
+                "The output '{output_name}' for flow is incorrect. The output value is not JSON serializable. "
+                "JSON dump failed: {error_type_and_message}. Please verify your flow output and "
```

Comment:
We don't need `.` after {error_type_and_message} since we assume the error_type_and_message already contains a `.` inside.

Reply:
why we assume that?

Reply:
Because our principle is that "the error message should be a natural language in English. Including correct capitalization, punctuation.".
{error_type_and_message} comes from the inner exception, which should also be a valid English sentence, which should have `.` at the end.